irohad/main/irohad.cpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "irohad.hpp"
+
+namespace iroha {
+  void Irohad::run() {
+    // run peer service
+    state_->peerService->run();
+
+    // temp solution
+    grpc::ServerBuilder builder;
+    builder.AddListeningPort(listen_on, grpc::InsecureServerCredentials());
+    builder.RegisterService(&server_);
+    auto server(builder.BuildAndStart());
+    // end
+
+    console_->info("Current order: {}", state_->peerService->orderToString());
+    console_->debug("started");
+
+    loop_->run();
+  }
+
+  void Irohad::bind_callbacks() {
+    // temp solution
+    server_.on<uvw::ErrorEvent>([c{console_}](const uvw::ErrorEvent& e,
+                                              auto& s) {
+      c->error("{}", e.what());
+
+      // TODO: we need better exception management
+      throw std::system_error();
+    });
+
+    std::shared_ptr<Member> state{state_};
+
+    server_.on<Proposal*>(
```

Comment:
The following code here is not only hard to read, but there are no comments explaining what it does.